GDAXSharp/Services/AbstractService.cs

```diff
@@ -122,12 +123,23 @@ public abstract class AbstractService
             return pagedList;
         }
 
-        protected static string SerializeObject(object value)
+        protected async Task<T> MakeServiceCall<T>(
+            HttpMethod httpMethod,
+            string uri,
+            string content = null)
+        {
+            var httpResponseMessage = await SendHttpRequestMessageAsync(httpMethod, uri, content);
```

Comment:
Not connected to this set of changes... but SendHttpRequestMessageAsync performs an httpClient.ReadAsStringAsync - so get the content body... Then the next line here does it again.

I know in WebAPI world a read on the content body of httpResponseMessage is forward only, meaning it can only be read once - so I was surprised its even possible to read it twice here... anyway, it's read twice. Maybe SendHttpRequestMessageAsync should return a tuple of the requestMessage and contentBody? Or just the content body? Or even just the already deserialized content body, as we know the Type we want back from it...

In similar area SendHttpRequestMessagePagedAsync<T> should really use SendHttpRequestMessageAsync  to make the request - as SendHttpRequestMessageAsync  checks for Status of 200 (SendHttpRequestMessagePagedAsync blows up when it gets non 200 responses as it can't deserialize the content body)

So what I'm suggesting here centralized the actual calls into one place - allowing for robust error handling and potential for logging...

Reply:
@chrisw000 I like your ideas.  will tackle it separately 👍